Source/DafnyCore/CoverageReporter/CoverageReport.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Linq;
+using Microsoft.Dafny;
+
+namespace DafnyCore.CoverageReporter; 
+
+public class CoverageReport {
+
+  private static int nextUniqueId = 0;
+
+  // INVARIANT: CoverageSpans are sorted within each list by the position of the StartToken
+  private readonly Dictionary<string, List<CoverageSpan>> labelsByFile;
+  public readonly string Name; // the name to assign to this coverage report
+  public readonly string Units; // the units of coverage (plural). This will be written in the coverage report table.
+  private readonly int uniqueId = nextUniqueId++;
```

Comment:
I think this isn't thread-safe, and I imagine we could at some point run multiple tests concurrency in a single process that would generate coverage reports.

Consider using `Interlocked.Increment`

Reply:
Done!